src/View/Helper/FormHelper.php

```diff
@@ -163,7 +163,9 @@ class FormHelper extends Helper
             'textarea' => '<textarea name="{{name}}"{{attrs}}>{{value}}</textarea>',
             // Container for submit buttons.
             'submitContainer' => '<div class="submit">{{content}}</div>',
-        ]
+        ],
+        // set HTML5 validation message to custom required/empty messages
+        'useValidationMessages' => false,
```

Comment:
Maybe rename it to `'injectHtml5ValidationMessages' or `injectClientValidationMessages` or `autoSetCustomValidity` or ... *something better?* ... basically anything not as ambiguous/unspecific as `useValidationMessages`, because validation messages are already being used by default (server side), and setting this option to false won't stop those.

Then:
It needs to be documented to enable this?
Will this be enabled by default in 4.x?

Reply:
I'm okay with renaming it; the setting is in the FormHelper context and it's annotated with a comment, so I didn't think it was too ambiguous, but I see what you mean. Of your choices, `autoSetCustomValidity` feels best to me (inject sounds danger danger).

It'll be enabled by default in 4.x if this gets merged. I will document it as well.